libr/core/cmd_anal.c

```diff
@@ -149,6 +152,54 @@ static int var_cmd (RCore *core, const char *str) {
 			}
 		}
 		break;
+	case 'd': {
+		char *name = r_str_chop(strdup(str+2));
+		RAnalVar *v = r_anal_var_get_byname (core->anal, fcn, type, name);
+		if (!v) {
+			eprintf ("no arg/local with this name exists\n");
+			free (name);
+			break;
+		}
+		char *fmt = r_anal_type_format (core->anal, v->type);
+		if (!fmt) {
+			 eprintf ("type:%s doesn't exist\n", v->type);
+			 break;
+		}
+		switch (type) {
+		case 'r':{
+				RRegItem *i = r_reg_index_get (core->anal->reg, v->delta);
+				if (!i) {
+					eprintf ("register not found\n");
+					break;
+				}
+				r_core_cmdf (core, "pf r (%s)\n", i->name);
```

Comment:
what about r_cons_printf instead of using core_cmdf? well that was my idea for the advd*, so maybe you can just cook the string with r_str_newf and then call printf or cmd depending on the last char of the command